js/index.js

```diff
@@ -1,304 +0,0 @@
-const buttons = document.getElementById("buttons-div");
-buttons.style.display="none";
-
-// setting up of the character selection buttons 
-function setPlayer(selectedPlayer){
-  player = selectedPlayer;
-  introScreen.style.display = "none";
-  buttons.style.display="flex";
-  screenshottext.style.display="block";
-  canvas.show();
-  loop();
-  counterInterval = setInterval(counter, 1000);
-  // play main song
-  mainSong.play();
-  mainSong.loop = true;
-  mainSong.volume = 0.2;
-}
-
-const player1btn = document.getElementById("player1");
-player1btn.onclick = function(){setPlayer(player1)};
-
-const player2btn = document.getElementById("player2");
-player2btn.onclick = function(){setPlayer(player2)};
-
-const player3btn = document.getElementById("player3");
-player3btn.onclick = function(){setPlayer(player3)};
-
-// setting up new game button & resetting the game 
-const newGame = document.getElementById("new-game-button");
-newGame.onclick = function(){
-  if (player === undefined) 
-    {player=player1} 
-        introScreen.style.display = "none";
-        buttons.style.display="flex";
-        screenshottext.style.display="block";
-        canvas.show();
-        score=0;
-        count=30;
-        scoreGoal = 100;
-        fill(255,255,255);
-        // reset bugs and skills , char position
-        x = 0;
-        y = 0;
-        fallingSpeed = 3;
-        bugs = [];
-        bugs2 = [];
-        skills = [];
-        skills2 = [];
-        bugsCount = 0;
-        bugs2Count = 0;
-        skillsCount = 0;
-        skills2Count = 0;
-        bugFrequency = 0.02;
-        bug2Frequency = 0;
-        skill2Frequency = -0.005;
-        // new counter
-        clearInterval(counterInterval);
-        counterInterval = setInterval(counter, 1000);
-        // restart the loop
-        loop();
-}
-
-// setting up the counter 
-let counter = function() {
-  count = count - 1; 
-  if (count <= 0) {
-    clearInterval(counterInterval);
-    count = 0;
-  }
-};
-
-// canvas setup function 
-function setup() {
-    canvas = createCanvas(1000,601);
-    canvas.hide();
-    textSize(width/30);
-    fill(255,255,255);
-    textFont('Andale Mono');
-    noLoop();
-}
-
-// canvas preload  function 
-function preload() {
-    bg = loadImage("img/bg-black.jpg");
-    player1 = loadImage("img/char1.webp");
-    player2 = loadImage("img/char2.webp");
-    player3 = loadImage("img/char5.webp");
-}
-
-// canvas draw function 
-function draw() {
-  background(220);
-  image(bg, 0, 0,);
-
-  if (player === player1) {
-    image(player1, 30 + x, 450 +y, 50,150);
-  }
-  if (player === player2) {
-    image(player2, 30 + x, 450 +y, 50,150);
-  }
-  if (player === player3) {
-    image(player3, 30 + x, 450 +y, 50,150);
-  }
-
-  drawObjects(Bug,bugs,bugFrequency);
-  drawObjects(Bug2,bugs2,bug2Frequency);
-  drawObjects(Skill,skills,0.02);
-  drawObjects(Skill2,skills2,skill2Frequency);
-
-  function drawObjects(ObjectClass, objectArray, frequency){
```

Comment:
I like this idea of encapsulating the draw objects functionality with the class as an argument like you did here.